app/src/main/cpp/Device.h

```diff
@@ -32,6 +32,43 @@ typedef int32_t DeviceType;
 const DeviceType UnknownType = 0;
 const DeviceType OculusGo = 1;
 const DeviceType OculusQuest = 2;
+
+enum class VRControllerType : uint8_t {
+  _empty,
+  HTCVive,
+  HTCViveCosmos,
+  HTCViveFocus,
+  HTCViveFocusPlus,
+  MSMR,
+  ValveIndex,
+  OculusGo,
+  OculusTouch,
+  OculusTouch2,
+  PicoGaze,
+  PicoG2,
+  PicoNeo2,
+  _end
+};
```

Comment:
I think you should just be adding new devices to `DeviceType` above.

Reply:
And only add the ones we support.

Reply:
How about I merge `DeviceType` into `VRControllerType`?  I would hope those types can mirror with VRControllerType with moz_external_vr.h because that will affect the controller button/axis conversion in Gecko.

Reply:
I think you need to coordinate with @MortimerGoro as he is using the DeviceType in the interstitial PR (#3136). I don't want to have devices in the list that we do not support. Rather than trying to keep DeviceType.java and Device.h and moz_external_vr.h in sync, I would rather have one internal set of values and then map them to gecko values in ExternalVR.cpp.

Reply:
I can understand. I will wait for his PR land, and we need to be sure give placeholder slots for the existing FF desktop controller types. The order are fixed.

Reply:
We don't need place holders. You will need to map them as we do for other values: https://github.com/MozillaReality/FirefoxReality/blob/717efee98f3556bb315190e45d5341b5a6ab28dd/app/src/main/cpp/ExternalVR.cpp#L231

Reply:
I already give a quick fix and I will merge once #3136 is landed.